src/lib/linux-api/src/lib.rs

```diff
@@ -5,9 +5,24 @@
 //! libc headers, to work with these kernel types. Normally this can't be done
 //! because the libc and kernel headers contain incompatible definitions.
 //!
-//! To support this use-case, we prefix most names with `Linux` or `linux_`
+//! # Type names
+//!
+//! To support the re-exported C bindings, we prefix most names with `Linux` or `linux_`
 //! as appropriate, so that the re-exported types don't conflict with the libc
-//! types that would otherwise have the same names.
+//! types that would otherwise have the same names. These types should be used in C
+//! APIs to avoid name conflicts.
+//!
+//! We also provide aliases or wrappers without the `linux_` prefix. These are
+//! bitwise compatible (transmutable) with the `linux_` versions of the types.
+//! We use wrappers in some cases; e.g.:
+//! * When the generated type is aliased to a primitive type (e.g. u64), which
+//!   would otherwise prevent us from implementing traits or methods for it.
+//! * When the generated type has bindgen-generated field names that shouldn't
+//!   be accessed directly outside of this crate.
+//!
+//! We also provide some types that are not necessarily bitwise-compatible with
+//! their linux counterparts, but are convertible. e.g. flag and enum types.
+//! Such types are given Rust-idiomatic camel-cased names.
```

Comment:
Optional: I found this a little difficult to follow on the first read through, and wasn't really sure what the takeaway was. If I'm reading this correctly, it seems like there's three categories of types exposed by this library; (1) aliases, (2) newtype wrappers, and (3) rust-only types. Maybe this documentation would be simpler if you listed these three categories and specified for each how they're named (so that the user can identify which type they are; is there a distinction in naming here?), what their properties are (ex: wrappers are transparent wrappers so are transmutable, "rust-only" types aren't, etc), and any other details.

Reply:
Yeah, good idea. Done.